src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+enum FilterType {
+  Active = 'active',
+  Completed = 'completed',
+  All = 'all',
+}
+
+const filterTodos = (todos: Todo[], filterType: string, filterText: string) => {
+  let filteredTodos: Todo[] = [...todos];
+
+  if (filterType === FilterType.Active) {
+    filteredTodos = filteredTodos.filter(todo => {
+      return !todo.completed;
+    });
+  }
+
+  if (filterType === FilterType.Completed) {
+    filteredTodos = filteredTodos.filter(todo => {
+      return todo.completed;
+    });
+  }
+
+  if (filterText) {
+    filteredTodos = filteredTodos.filter(todo =>
+      todo.title.toLowerCase().includes(filterText.toLowerCase()),
+    );
+  }
+
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodods] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [filterType, setFilterType] = useState<string>('All');
+  const [filterText, setFilterText] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos().then(data => {
+      setTodods(data);
+      setIsLoading(false);
```

Comment:
What if there will be an error? 
Good idea with adding loaders, but to make this usage fully valid lets move setting loading to false at the `finally` block and add an error catcher.